### PR TITLE
AV-5576: Replace zipruby gem with rubyzip

### PR DESCRIPTION
**Pull request by khanh227**
CI can not automatically run, so this is my local spec result. 
image

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,3 +9,6 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [2.2.1] - 2022-04-07
 ### Fixed
 - Override `#enrolment_store` from base class in `AvetmissData::Stores::V8::QualCompletion` to match the correct enrolment store in cases where there are multiple enrolments for a client identifier
+
+## [3.0.0] - 2022-10-25
+### Replace zipruby gem with rubyzip
```
Comment (schlick) [inline CHANGELOG.md:14]: Move this between Unreleased and 2.2.1. New entries go at the top of the list

**Review by schlick — CHANGES_REQUESTED** [review]
Just the one minor change please

File: lib/avetmiss_data/package.rb [hunk 2]
```diff
@@ -174,10 +174,10 @@ def initialize_stores_list(list)
...
 
   def generate_zip_file(files_list)
-    Zip::Archive.open_buffer(Zip::CREATE) do |archive|
+    Zip::File.open_buffer(create: true) do |archive|
       files_list.each_pair do |stores_name, file_name|
         lines = send(stores_name).map { |store| store.to_line }.join("\n")
-        archive.add_buffer("#{file_name}.txt", lines)
+        archive.add("#{file_name}.txt", lines)
       end
     end
```
Comment (schlick) [inline lib/avetmiss_data/package.rb:182]: I played around with it and banged my head many times. I ended up with this:

```
    Zip::File.add_buffer do |archive|
      files_list.each_pair do |stores_name, file_name|
        lines = send(stores_name).map { |store| store.to_line }.join("\n")
        archive.get_output_stream("#{file_name}.txt") { |os| os.write lines }
      end
    end
```

File: lib/avetmiss_data/zip_file.rb [hunk 3]
```diff
@@ -38,7 +38,7 @@ def build_stores(file_name, lines)
   def stores
     @stores = {}
     read_archive do |file|
-      content = file.read
+      content = file.get_input_stream.read
       name = store_name_for(file.name)
       if content.nil?
```
Comment (schlick) [inline lib/avetmiss_data/zip_file.rb:43]: I also had to make this change:

```
-      if content.nil?
+      if content.blank?
```

File: lib/avetmiss_data/package.rb (hunk 2 above)
Comment (schlick) [reply lib/avetmiss_data/package.rb:182]: But I also had to change the line mentioned in the other comment for zip_file.rb and also these lines in package_spec.rb:

```
-        File.open(temp_file.path, 'wb:ASCII-8BIT') { |f| f << package.to_zip_file }
+        File.open(temp_file.path, 'wb:ASCII-8BIT') { |f| f << package.to_zip_file.string }
```
and
```
-        File.open(temp_file.path, 'wb:ASCII-8BIT') { |f| f << package.to_aggregate_zip_file }
+        File.open(temp_file.path, 'wb:ASCII-8BIT') { |f| f << package.to_aggregate_zip_file.string }
```

Comment (schlick) [reply lib/avetmiss_data/package.rb:182]: But I don't know if this change actually does what it is meant to achieve. I'm not convinced the specs are totally covering it.

**Review by schlick — CHANGES_REQUESTED** [review]

**Comment by khanh227** [comment]
Thank @schlick for your help to review and suggestion, spec case fixed at last changes. 
image

**Review by schlick — APPROVED** [review]

**Pull request merged**